module/icp/algs/impl/impl.c

```diff
@@ -0,0 +1,175 @@
+
+#include <sys/simd.h>
+#include <impl/impl.h>
+
+#define	ALG_IMPL_READ(i) (*(volatile uint32_t *) &(i))
+
+static const struct {
+	char *name;
+	uint32_t sel;
+} alg_impl_opts[] = {
+	{ "cycle", ALG_IMPL_CYCLE },
```

Comment:
Who uses `cycle`?

Reply:
I believe cycle is commonly used in [ztest](https://github.com/openzfs/zfs/blob/master/cmd/ztest/ztest.c#L4221)

Reply:
That's right.  We use cycle in ztest as mentioned in the comment to rotate through all the supported algorithms.  This helps ensure we're testing them all and that each of them produces the same result.  If they didn't we're quickly see problems.

Reply:
Ok - will leave this in, and add it to ztest.